src/main/java/de/tum/in/test/api/structural/StructuralTestProvider.java

```diff
@@ -379,7 +379,10 @@ public JSONObject getExpectedClassJson() {
 		}
 
 		public String getQualifiedClassName() {
-			return expectedPackageName + "." + expectedClassName;
+			if (expectedPackageName != null && !expectedPackageName.isEmpty()) {
```

Comment:
```suggestion
			if (!expectedPackageName.isEmpty()) {
```
The `expectedPackageName` is final here and set using `Objects.requireNonNull`, so it cannot be null. I think we can remove this check. Otherwise, we need to change the constructor as well and check all usages of `getExpectedPackageName()`.